deploy/buildtools/buildglobal.py

```diff
@@ -0,0 +1,105 @@
+""" This converts the build configuration files into something usable by the
+manager """
+
+from time import strftime, gmtime
+import ConfigParser
+import pprint
+from fabric.api import *
+import logging
+
+from runtools.runtime_config import RuntimeHWDB
+from buildtools.buildconfig import BuildConfig
+from buildtools.buildlocal import GlobalLocalBuildConfig
+from buildtools.buildafi import GlobalAwsBuildConfig
+
+rootLogger = logging.getLogger()
+
+class GlobalBuildConfig:
+    """ Configuration class for builds. This is the "global" configfile,
+    i.e. sample_config_build.ini
+    """
+
+    def __init__(self, args):
+        if args.launchtime:
+            launch_time = args.launchtime
+        else:
+            launch_time = strftime("%Y-%m-%d--%H-%M-%S", gmtime())
+
+        self.args = args
+
+        global_build_configfile = ConfigParser.ConfigParser(allow_no_value=True)
+        # make option names case sensitive
+        global_build_configfile.optionxform = str
+        global_build_configfile.read(args.buildconfigfile)
+
+        builds_to_run_list = map(lambda x: x[0], global_build_configfile.items('builds'))
+
+        build_recipes_configfile = ConfigParser.ConfigParser(allow_no_value=True)
+        # make option names case sensitive
+        build_recipes_configfile.optionxform = str
+        build_recipes_configfile.read(args.buildrecipesconfigfile)
+
+        self.builds_list = []
+        for build_name in builds_to_run_list:
+            self.builds_list.append(BuildConfig(
+                build_name,
+                dict(build_recipes_configfile.items(build_name)),
+                launch_time))
+
+        self.hwdb = RuntimeHWDB(args.hwdbconfigfile)
+
+        # setup the particular host_platform
+        self.host_platform = global_build_configfile.get('buildfarmconfig', 'hostplatform')
+        if self.host_platform == "aws":
+            self.host_platform_config = GlobalAwsBuildConfig(args, global_build_configfile)
+        elif self.host_platform == "local":
+            self.host_platform_config = GlobalLocalBuildConfig(args, global_build_configfile, self.builds_list)
+        else:
+            sys.exit('Invalid host platform: {}'.format(host_platform))
```

Comment:
#763 would help this type of code too.  If we merge it, uses of sys.exit like this would get logged.  Right now, they wouldn't.